### PR TITLE
TIP-687: Make the versioning great again

### PR DESCRIPTION
**Pull request by SamirBoulil**
**Description (for Contributor and Core Developer)**
- Removes ProductValueUpdateGuesser as it is unused anymore
- Add integration tests on the version creation for products

**Definition Of Done (for Core Developer only)**

| Q                                 | A
| --------------------------------- | ---
| Added Specs                       | -
| Added Behats                      | -
| Added integration tests           | OK
| Changelog updated                 | OK
| Review and 2 GTM                  | Todo
| Micro Demo to the PO (Story only) | -
| Migration script                  | -
| Tech Doc                          | -

`Todo`: Pending / Work in progress
`OK`: Done / Validated
`-`: Not needed

### CODE REVIEW

File: src/Pim/Bundle/VersioningBundle/Resources/config/guessers.yml [hunk 1]
```diff
@@ -4,7 +4,6 @@ parameters:
     pim_versioning.update_guesser.chained.class:           Pim\Bundle\VersioningBundle\UpdateGuesser\ChainedUpdateGuesser
     pim_versioning.update_guesser.contains_products.class: Pim\Bundle\VersioningBundle\UpdateGuesser\ContainsProductsUpdateGuesser
     pim_versioning.update_guesser.variant_group.class:     Pim\Bundle\VersioningBundle\UpdateGuesser\VariantGroupUpdateGuesser
-    pim_versioning.update_guesser.product_value.class:     Pim\Bundle\VersioningBundle\UpdateGuesser\ProductValueUpdateGuesser
```
Comment (SamirBoulil) [inline src/Pim/Bundle/VersioningBundle/Resources/config/guessers.yml:7]: Is removing a parameter a BC that should be added to the changelog, or the service definition is enough ?

Comment (damien-carcel) [inline src/Pim/Bundle/VersioningBundle/Resources/config/guessers.yml]: IMO, we should do both. The class could have been removed, but the service changed to use a new one, for instance.

Comment (damien-carcel) [reply src/Pim/Bundle/VersioningBundle/Resources/config/guessers.yml:7]: From 1.7 changelog:
`- Remove useless service and parameter: 'pim_enrich_image' and 'pim_enrich.form.type.image.class'`
So I would say: note it all!

File: src/Pim/Bundle/VersioningBundle/tests/integration/Manager/VersionManagerIntegration.php [hunk 2]
```diff
@@ -0,0 +1,183 @@
...
+    }
+
+    public function testCreateProductVersionOnAttributeAndFieldDeletion()
+    {
+        $product = $this->productRepository->findOneByIdentifier('versioned-product');
+        $productValue = $product->getValue('a_date');
+        $product->removeValue($productValue);
+        $this->productSaver->save($product);
+
+        $productVersions = $this->get('pim_versioning.repository.version')
```
Comment (jjanvier) [inline src/Pim/Bundle/VersioningBundle/tests/integration/Manager/VersionManagerIntegration.php]: You could init the service in the setup

**Review by jjanvier — APPROVED** [review]
Great! GTM if this test is green.
Are the behat OK or is still impossible for those ones?

File: src/Pim/Bundle/VersioningBundle/tests/integration/Manager/VersionManagerIntegration.php (hunk 2 above)
Comment (SamirBoulil) [reply src/Pim/Bundle/VersioningBundle/tests/integration/Manager/VersionManagerIntegration.php]: Yep I did, just didn't use the private here. I fix it.

**Comment by SamirBoulil** [comment]
@jjanvier Nope, I can't run the behats for the products (The PEF is still not shown), but the behats for simple entities are running (and green from what I've seen on my machine).

**Comment by SamirBoulil** [comment]
The Integration test is green on CE and EE.

**Comment by SamirBoulil** [comment]
Travis is red on Php7.0 because https://github.com/akeneo/pim-community-dev/blob/master/src/Pim/Bundle/VersioningBundle/Doctrine/ORM/VersionRepository.php#L31 does not sort properly the versions it returns.
The quick fix would be to look for the last version by hand or even worse sort the result.. but that would break the integration test.
Or not running statefull tests :/

Instead we should find a way to fix this sorting issue in php7.0

**Comment by ClementGautier** [comment]
IMO it's not directly related to php7, you could have the same behavior on php5 with a powerfull enough computer. Can't you just put a wait of some milliseconds between the creation of the 2 versions in the test ?

**Comment by SamirBoulil** [comment]
@ClementGautier well the database precision is seconds, putting 1s sleep makes the behavior dissapear forever. Sleep of milliseconds is not worth of an optimization.

**Review by phaseinducer — APPROVED** [review]

**Review by damien-carcel — APPROVED** [review]

**Comment by ClementGautier** [comment]
1s precision ? :scream: This is so long even a human could make 2 versions appears to be in a bad order. 
In my opinion we are in a case where the test tells you to review the design because there is obviously a bug here. I'm not saying that the PR is bad, (I give you my GTM by the way if you make the test pass in any way you find), I'm just saying that, by design here versions can be miss-ordered.

**Review by ClementGautier — APPROVED** [review]

**Pull request merged**